### PR TITLE
scalebar component implemented

### PR DESCRIPTION
**Pull request by mapolovina**

### CODE REVIEW

**Comment by mbarto** [comment]
Hi, Marko,as you can see, your pull request is not currently building on Travis CI. 
Look here: https://travis-ci.org/geosolutions-it/MapStore2/builds/79132444. 
We use ESLint to check for common errors. You can run a test locally with npm run lint. Please fix all the reported issues. We also use plugins in our Editor to check rules as we develop. You can find plugins for Atom, Sublime or many other errors that can help you. They will automatically read the .eslintrc in the project.

**Comment by mapolovina** [comment]
Hi Mauro, I hope that now it's ok.
Sincerely,
Marko.

**Comment by simboss** [comment]
Marko, I doubt things are fine, the PR is not even mergeable.

A clean work should end up with a mergeable PR. I also see some travis, failures ( I did no investigated deep though).

Let see what Mauro thinks.

**Comment by mapolovina** [comment]
Sorry about merge, solved.

**Comment by simboss** [comment]
Still Red Light :(

You did not provide enough unit test. PR are not mergeable until coveralls and travis are both happy.

File: web/client/api/ScaleBox.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ */
+
+/**
+ * [ScaleBox component]
+ * @param {[Boolean]} showBar   [if you want to display scalebar]
+ * @param {[Boolean]} showCombo [if you want to display scalecombo]
+ * @param {[String]} units [units as string, eg. 'm', 'km', 'mi']
+ * @param {[Object]} options   [component options]
+ */
+var ScaleBox = function(map, showBar, showCombo, units, options) {
```
Comment (mbarto) [inline web/client/api/ScaleBox.js]: React components should be self contained, api utilities should be used for general functionality, not for UI purposes

File: web/client/components/ScaleBar/ScaleBar.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+     */
+    getInitialState() {
+        return {};
+    },
+    /**
+     * [componentDidMount after component is loaded]
+     * @return {[undefined]} [no return]
+     */
+    componentDidMount() {
+        var that = this;
```
Comment (mbarto) [inline web/client/components/ScaleBar/ScaleBar.jsx]: I don't personally like the use of that, me and so on. There are better ways to bind functions to this. For example using ES6 arrow functions, or explicity using the bind method.

File: web/client/components/ScaleBar/ScaleBar.jsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+var React = require('react');
+
+/**
+ * [ScaleBarController new React Controller for ScaleBar]
+ */
+var ScaleBarController = React.createClass({
```
Comment (mbarto) [inline web/client/components/ScaleBar/ScaleBar.jsx]: Component name should be equal to file name. Moreover this is not a controller, but a Component. In general ScaleBar is a good name.

File: web/client/components/ScaleBar/ScaleBox.jsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+     * @return {[Object]} [return scalebox object]
+     */
+    getDefaultProps() {
+        return {
+            scalebox: {
+                combo: false,
+                bar: true,
+                units: 'm',
+                getComboItems: function() {
+                    var items = [[18, '18'], [17, '17'], [16, '16'], [15, '15'], [14, '14'], [13, '13'], [12, '12'], [11, '11'], [10, '10'], [9, '9'], [8, '8'], [7, '7'], [6, '6'], [5, '5'], [4, '4'], [3, '3'], [2, '2'], [1, '1'], [0, '0']];
```
Comment (mbarto) [inline web/client/components/ScaleBar/ScaleBox.jsx]: What if I want 1000 levels of zoom?

File: web/client/components/ScaleBar/ScaleBox.jsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+var React = require('react');
+var ScaleComboController = require('./ScaleCombo');
+var ScaleBarController = require('./ScaleBar');
+
+/**
+ * [ScaleBoxController new React Controller for ScaleBox]
```
Comment (mbarto) [inline web/client/components/ScaleBar/ScaleBox.jsx]: React Controllers do not exist. React is not MVC, is just View.

File: web/client/components/ScaleBar/ScaleCombo.jsx [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+    },
+    /**
+     * [getDefaultProps fetch default props]
+     * @return {[Object]} [return scalebox object]
+     */
+    getDefaultProps() {
+        return {
+            scalebox: {
+                getComboItems: function() {
+                    var items = [[18, '18'], [17, '17'], [16, '16'], [15, '15'], [14, '14'], [13, '13'], [12, '12'], [11, '11'], [10, '10'], [9, '9'], [8, '8'], [7, '7'], [6, '6'], [5, '5'], [4, '4'], [3, '3'], [2, '2'], [1, '1'], [0, '0']];
```
Comment (mbarto) [inline web/client/components/ScaleBar/ScaleCombo.jsx]: What if I want 1000 levels of zoom?

File: web/client/components/ScaleBar/ScaleCombo.jsx [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+    },
+    /**
+     * [render render view]
+     * @return {[HTML element]} [html]
+     */
+    render() {
+        return (
+            <div>
+                <DropdownButton className="mapstore-scalebox-combo-main" style={this.state.showCombo} title={this.state.activeVal} dropup pullRight>
+                {this.state.items()}
```
Comment (mbarto) [inline web/client/components/ScaleBar/ScaleCombo.jsx]: I would prefer a more declarative approach. State is not needed here. In general we prefer stateless components wherever possible.

File: web/client/components/leaflet/Map.jsx [hunk 8]
```diff
@@ -7,6 +7,11 @@
  */
 var L = require('leaflet');
 var React = require('react');
+var ScaleBox = require('../../api/ScaleBox');
```
Comment (mbarto) [inline web/client/components/leaflet/Map.jsx]: Map should not depend on optional components. Components inclusion should be done by composition on a higher level container.

**Pull request closed without merging**